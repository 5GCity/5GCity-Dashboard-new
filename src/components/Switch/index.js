/**
 * Switch Component
 * Please write a description
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import React from 'react'
import styled from 'styled-components'
import { Switch } from 'element-react'

export default ({ children, title, ...props }) => (
  <Wrapper >
    {Title &&
    <Title>{title}</Title>
  }
    <SwitchContainer {...props} />
  </Wrapper>
)

const Wrapper = styled.div``

const Title = styled.span`
    color:white;
    margin-right: 8px;
    font-size: 12px;
    line-height: 12px;
    font-family: ${({ theme }) => theme.fontFamily};
`

const SwitchContainer = styled(Switch)`


.el-switch__button {

}

.el-switch__core {
 position: unset;
 background: transparent;
 boder-radius: 7px;
 border:none;
/*  width: 28px !important; */


.el-switch__label .el-switch__label--right {
 opacity:0.2;
}
 height: 12px;


.el-switch__label .el-switch__label--right {
 opacity:0.2;
}
}


.el-switch__label .el-switch__label--right {
 opacity:0.2;
}`
