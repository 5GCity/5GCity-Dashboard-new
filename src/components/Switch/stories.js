/**
 * Switch Component Stories
 * Please write a description or remove this line
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'
import { Theme } from 'globalStyles'

import Switch from './index'

const bStyle = {
  padding: '30px',
  backgroundColor: Theme.bodyBackground,
  height: '100%'
}

const example = () => (
  <div style={bStyle}>
    <Switch title={'Show bandwidth'} name='test switch' onText='' offText='' onColor='#8CC14E' offColor='#89979F' />
    <Switch name='test switch' onText='' offText='' onColor='#8CC14E' offColor='#89979F' />
  </div>
)

storiesOf('Switch', module)
  .add('default', example)
