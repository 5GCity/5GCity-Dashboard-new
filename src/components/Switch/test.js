/**
 * Switch Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Switch from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<Switch />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
