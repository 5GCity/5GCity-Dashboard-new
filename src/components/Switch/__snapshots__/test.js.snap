// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  color: white;
  margin-right: 8px;
  font-size: 12px;
  line-height: 12px;
}

.c1 .el-switch__core {
  position: unset;
  background: transparent;
  boder-radius: 7px;
  border: none;
  height: 12px;
}

.c1 .el-switch__core .el-switch__label .el-switch__label--right {
  opacity: 0.2;
}

.c1 .el-switch__core .el-switch__label .el-switch__label--right {
  opacity: 0.2;
}

.c1 .el-switch__label .el-switch__label--right {
  opacity: 0.2;
}

<div
  className=""
>
  <span
    className="c0"
  />
  <label
    className="el-switch el-switch--wide is-checked c1"
    style={Object {}}
  >
    <div
      className="el-switch__mask"
      style={
        Object {
          "display": "none",
        }
      }
    />
    <input
      checked={true}
      className="el-switch__input c1"
      disabled={false}
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <span
      className="el-switch__core"
      style={
        Object {
          "width": "58px",
        }
      }
    >
      <span
        className="el-switch__button"
        onClick={[Function]}
        style={
          Object {
            "transform": "translate(38px, 2px)",
          }
        }
      />
    </span>
    <div
      className="el-switch__label el-switch__label--left"
      style={
        Object {
          "width": "58px",
        }
      }
    >
      
      <span>
        ON
      </span>
    </div>
    <div
      className="el-switch__label el-switch__label--right"
      style={
        Object {
          "display": "none",
          "width": "58px",
        }
      }
    >
      
      <span>
        OFF
      </span>
    </div>
  </label>
</div>
`;
